test/asyncio_tests/__init__.py

```diff
@@ -95,23 +95,31 @@ def setUp(self):
         self.collection = self.db.test_collection
         self.loop.run_until_complete(self.collection.drop())
 
-    def get_client_kwargs(self, **kwargs):
+    def _get_ssl_kwargs(self, **kwargs):
         if env.mongod_started_with_ssl:
             kwargs.setdefault("tlsCAFile", CA_PEM)
             kwargs.setdefault("tlsCertificateKeyFile", CLIENT_PEM)
-
         kwargs.setdefault("tls", env.mongod_started_with_ssl)
-        kwargs.setdefault("io_loop", self.loop)
+        return kwargs
 
+    def _get_io_loop_kwargs(self, **kwargs):
+        kwargs.setdefault("io_loop", self.loop)
         return kwargs
 
+    def get_client_kwargs(self, **kwargs):
+        return self._get_io_loop_kwargs(**self._get_ssl_kwargs(**kwargs))
+
     def asyncio_client(self, uri=None, *args, **kwargs):
         """Get an AsyncIOMotorClient.
 
         Ignores self.ssl, you must pass 'ssl' argument.
         """
+        if kwargs.get("host", None):
+            kwargs = self._get_ssl_kwargs(**kwargs)
+        else:
+            kwargs = self.get_client_kwargs(**kwargs)
```

Comment:
It does not make sense to base this if/else on the presence of the "host" kwarg. Let's always use get_client_kwargs here and figure out a more straightforward way to avoid adding the loop parameter in for test_shared_client.